services/wfm/src/main/java/org/openkilda/wfm/topology/stats/FlowDirectionHelper.java

```diff
@@ -28,25 +27,30 @@ private FlowDirectionHelper() {}
         REVERSE
     }
 
-
     /**
      * Trys to determine the direction of the flow based on the cookie.
      *
-     * @param cookie cookie
+     * @param rawCookie cookie
      * @return flow direction
      */
-    public static Direction findDirection(long cookie) throws FlowCookieException {
-        // Kilda flow first number represents direction with 4 = forward and 2 = reverse
-        // high order nibble represents type of flow with a 2 representing a forward flow
-        // and a 4 representing the reverse flow
-        if (!isKildaCookie(cookie)) {
-            throw new FlowCookieException(cookie + " is not a Kilda flow");
+    public static Direction findDirection(long rawCookie) throws FlowCookieException {
+        Cookie cookie;
+        try {
+            cookie = Cookie.decode(rawCookie);
+        } catch (IllegalArgumentException e) {
+            throw new FlowCookieException(e.getMessage());
         }
-        long direction = cookie & 0x6000000000000000L;
-        if ((direction != FORWARD_FLOW_COOKIE_MASK) && (direction != REVERSE_FLOW_COOKIE_MASK)) {
+
+        Direction direction;
+        if (cookie.isMaskedAsForward()) {
+            direction = Direction.FORWARD;
+        } else if (cookie.isMaskedAsReversed()) {
+            direction = Direction.REVERSE;
+        } else {
             throw new FlowCookieException("unknown direction for " + cookie);
         }
-        return direction == FORWARD_FLOW_COOKIE_MASK ? Direction.FORWARD : Direction.REVERSE;
+
+        return direction;
     }
 
     static boolean isKildaCookie(long cookie) {
```

Comment:
It is unused now

Reply:
Didn't get. What "unused now"?

Reply:
You removed using of `isKildaCookie()` method in line 42.

There is no other place where this method is used (except tests). 
So this method can be removed now

Reply:
Now I got it. My fail, removing it.